db/background_mgr_attachment_compaction.go

```diff
@@ -127,8 +128,28 @@ func (a *AttachmentCompactionManager) Run(ctx context.Context, options map[strin
 	case "mark", "":
 		a.SetPhase("mark")
 		persistClusterStatus()
-		_, a.VBUUIDs, err = attachmentCompactMarkPhase(ctx, dataStore, collectionID, database, a.CompactID, terminator, &a.MarkedAttachments)
+		worker := func() (shouldRetry bool, err error, value interface{}) {
+			_, a.VBUUIDs, err = attachmentCompactMarkPhase(ctx, dataStore, collectionID, database, a.CompactID, terminator, &a.MarkedAttachments)
+			if errors.Is(err, base.ErrAttachmentCompactionRollback) {
+				base.InfofCtx(ctx, base.KeyDCP, "rollback indicated on mark phase of attachment, resetting the task")
+				err = a.newRunRollback(ctx, options)
+				if err != nil {
+					base.WarnfCtx(ctx, "error on initialization of new run after rollback has been indicated")
+					return false, err, nil
+				}
+				// we should try again if it is rollback error
+				return true, nil, nil
+			}
+			// if error isn't rollback then assume its not recoverable
+			return false, err, nil
+		}
+		// retry loop for handling a rollback during mark phase of compaction process
+		err, _ = base.RetryLoop("attachmentCompactMarkPhase", worker, base.CreateMaxDoublingSleeperFunc(25, 100, 10000))
 		if err != nil || terminator.IsClosed() {
+			if errors.Is(err, base.ErrAttachmentCompactionRollback) {
+				// log warning to show we hit max number of retries
+				base.WarnfCtx(ctx, "maximum retry attempts reached on mark phase: %v", err)
```

Comment:
I think we should log when we purge the metadata, see above that this might occur in a different spot depending on how the error is handled.